functions/src/modules/SQR.functions.ts

```diff
@@ -303,118 +299,107 @@ export const processSubmission = functions.database
+        created: row['Created'] || null,
+        comments: row['Comments'],
         soundissues,
-        soundqualityrating: row['soundqualityrating'],
+        soundqualityrating: row['Sound Quality Rating'],
         unwantedparts,
         duration: {
-          beginning: new Date(row['beginning']).getTime() / 1000,
-          ending: new Date(row['ending']).getTime() / 1000,
-        },
+          beginning: row['Beginning'],
+          ending: row['Ending'],
+        }
       };
+      db.ref(`/submissions/soundQualityReporting/${list}/${audioFileName}/${token}`).set(submission);
+    }
 
-      db.ref(`/sqr/submissions/${serial}`).set(submission);
-    });
+    ////////////////////////
+    //     Allotments
+    ////////////////////////
 
-    const getAllotments = promisify(AllotmentsSheet.getRows);
-    // alllotments = Allotments sheet rows
-    const allotments = await getAllotments();
-
-    // Group all the files allotted on one day under a single `Allotment Node` in the db
-    // Group by ASSIGNEEs/DATEs/LISTs
-
-    const groupedAllotments = helpers.groupByMulti(
-      allotments,
-      ['devotee', 'dategiven', 'list'],
-      {}
-    );
-
-    // Adding the allotments
-    for (const assignee in groupedAllotments) {
-      for (const date in groupedAllotments[assignee]) {
-        const dayFiles = [];
-        for (const list in groupedAllotments[assignee][date]) {
-          // Collecting all of the files on a list under a single day in an array
-          groupedAllotments[assignee][date][list].forEach(item => {
-            dayFiles.push(item['filename']);
-          });
+    const allotmentsSheet = new GoogleSheets(spreadsheetId, ISoundQualityReportSheet.Allotments);
+    const allotmentsRows = await allotmentsSheet.__getRows();
 
-          const allotment = {
-            assignee: {
-              name: assignee,
-              emailAddress: groupedAllotments[assignee][date][list][0]['email'],
-            },
-            files: dayFiles,
-            list,
-            timestamp: new Date(date).getTime() / 1000,
-            sendNotificationEmail: false, // Don't send emails if the document is read from the spreadsheet
-          };
-
-          db.ref(`/sqr/allotments`).push(allotment);
-        }
+    for (const row of allotmentsRows) {
+      if (!row) continue;
+
+      const fileName = row['File Name'];
+      const match = audioFileRegex.exec(fileName);
+
+      if (!match) continue;
+
+      const fileNameHasForbiddenChars = fileName.match(/[\.\[\]$#]/g);
+      if (fileNameHasForbiddenChars) {
+        console.warn(`File name has forbidden characters that can't be used as a node name.`)
+        continue;
       }
-    }
 
-    res
-      .status(200)
-      .send(
-        `Function was called successfully, check the Logs on Firebase to find out if something went wrong`
-      );
-  }
-);
+      const list = match[1];
+      const allotment = {
+        status: row['Status'] || null,
```

Comment:
How does this `||` work here and what is the purpose?

Reply:
This is to set the value of `status` to null so that it isn't written to the db when the cell value is an empty string

Reply:
Got it.